packages/postcss-lowercase-text/src/index.js

```diff
@@ -0,0 +1,66 @@
+import * as postcss from 'postcss';
+import valueParser from 'postcss-value-parser';
+import unitTransformer from './units';
+import selectorTransformer from './selector';
+import atRulesTranformerMap from './atRules';
+
+export default postcss.plugin('postcss-lowercase-props-selectors', () => {
+  return (css) => {
+    let allAtRulesNames = []; // ignored list for transforming values
+    css.walkAtRules((rule) => {
+      const allowedTransformingTypes = Object.keys(atRulesTranformerMap);
+      const allowedTransformingTypeSet = new Set(allowedTransformingTypes);
+      if (allowedTransformingTypeSet.has(rule.name.toLowerCase())) {
+        atRulesTranformerMap[rule.name.toLowerCase()](rule);
+      }
+      // @font-face doesn't have params
+      if (typeof rule.params !== 'undefined') {
+        const params = rule.params; // need to copy the params as adding toLowerCase here will change the params in AST
+        allAtRulesNames.push(params.toLowerCase());
+        // its safe to use the lowercase to matching as only params of
+        // @counter-styles is being lowercase and they are case-insensitive
+      }
+    });
+
+    css.walkRules((rule) => {
+      rule.walkDecls((decl) => {
+        // handling properties
+        // All properties of CSS are case-insensitive. SAFE to transform
+
+        // we cant simply lowercase them as css variables are declared using props syntax only
+        if (!/^--(.)+/.test(decl.prop)) {
+          decl.prop = decl.prop.toLowerCase();
+        }
+        // font-family is case sensitive prop, its value should be left as it is
+        if (decl.prop === 'font-family') {
```

Comment:
We have many properties which we should lowercased

Reply:
can you name them ? may all `font` as well ?